slsReceiverSoftware/src/DataProcessor.cpp

```diff
@@ -316,16 +316,17 @@ uint64_t DataProcessor::ProcessAnImage(char *buf) {
     try {
         // normal call back
         if (rawDataReadyCallBack != nullptr) {
-            rawDataReadyCallBack((char *)rheader,
+            size_t dsize = (uint32_t)(*((uint32_t *)buf));
+            rawDataReadyCallBack(rheader,
                                  buf + FIFO_HEADER_NUMBYTES +
                                      sizeof(sls_receiver_header),
-                                 (uint32_t)(*((uint32_t *)buf)), pRawDataReady);
+                                 dsize, pRawDataReady);
         }
 
         // call back with modified size
         else if (rawDataModifyReadyCallBack != nullptr) {
-            auto revsize = (uint32_t)(*((uint32_t *)buf));
-            rawDataModifyReadyCallBack((char *)rheader,
+            size_t revsize = (uint32_t)(*((uint32_t *)buf));
```

Comment:
Would this be more clear? 
```
std::size_t revsize = *reinterpret_cast<uint32_t*>(buf);
```

Reply:
done for both the casts.. and static for the startacquisitioncallback imagesize